src/simulator.cpp

```diff
@@ -97,69 +98,57 @@ void Simulator::ReadConfig() {
 }
 
 void Simulator::RunSimulation() {
-    std::ifstream in_file((this->trace_file).c_str(), std::ios::in);
-    char trace_line[13];
-
-    if (in_file.fail()) {
-        std::cerr << "Open trace file error" << std::endl;
-        exit(-1);
-    }
-
-    while (!in_file.eof()) {
+    while (inst_loader->IfAvailable()) {
         try {
-            in_file.getline(trace_line, LENGTH_OF_INST_LINE);
-            bool is_success = _CacheHandler(trace_line);
+            bool is_success = _CacheHandler(inst_loader->GetNextInst());
             if (!is_success) {
                 throw std::logic_error("Cache Handler failed");
             }
         } catch (std::exception &ex) {
-            in_file.close();
             std::cerr << ex.what() << std::endl;
             exit(-1);
         }
     }
-    in_file.close();
     _CalHitRate();
 }
 
-bool Simulator::_CacheHandler(char *trace_line) {
+bool Simulator::_CacheHandler(inst_t inst) {
     bool is_load(false), is_store(false), is_space(false);
     // Determine what kind of the instruction
-    switch (trace_line[0]) {
-    case 'l':
+    switch (inst.op) {
+    case I_LOAD:
         is_load = true;
         break;
-    case 's':
+    case I_STORE:
         is_store = true;
         break;
-    case '\0':
+    case I_NONE:
         is_space = true;
         break;
-    default:
-        std::cerr << "Undefined instruction type." << std::endl;
-        std::cerr << "Error line: " << trace_line << std::endl;
-        return false;
     }
 
-    // Parse the address from the instruction
-    auto temp = strtoul(trace_line + 2, nullptr, INST_ADDR_BASE);
-    std::bitset<32> addr(temp);
+    addr_t addr_b = Cvt2AddrBits(inst.addr_raw);
 
-    // Preprocess the following memory address
-    std::bitset<32> poten_victim_addr = this->main_cache->Ready(addr);
+    std::bitset<32> poten_victim_addr = this->main_cache->Ready(addr_b);
 
     // Handle the address
     if (is_load) {
         ++_counter.access;
         ++_counter.load;
+<<<<<<< HEAD
```

Comment:
Brach history leftover.....

Reply:
Interesting.. I don't see any "HEAD" in my source code